system-report.sh

```diff
@@ -50,15 +50,30 @@ set_pkg_result_string() {
     result="$result\n$(apt info ${package} 2>/dev/null | grep -E 'Package|Version|APT-Sources')"
 }
 
+# load msr kernel module and install msr-tools if needed
+install_msr() {
+    if ! which rdmsr &> /dev/null; then
+        sudo apt install -y msr-tools &> /dev/null
+    fi
+    sudo modprobe msr &> /dev/null
+}
+
 set_msr_result_string() {
-    HW_ENCRYPT_ENABLE=$(sudo rdmsr 0x982 -f 1:1)
-    result="HW_ENCRYPT_ENABLE bit: ${HW_ENCRYPT_ENABLE} (expected value: 1)"
+    install_msr
+    MK_TME_ENABLE=$(sudo rdmsr 0x982 -f 1:1)
+    result="MK_TME_ENABLE bit: ${MK_TME_ENABLE} (expected value: 1)"
     SEAM_RR=$(sudo rdmsr 0x1401 -f 11:11)
     result="$result\nSEAM_RR bit: $SEAM_RR (expected value: 1)"
     NUM_TDX_PRIV_KEYS=$(sudo rdmsr 0x87 -f 63:32)
-    result="$result\nNUM_TDX_PRIV_KEYS: $NUM_TDX_PRIV_KEYS (expected value: >32)"
+    result="$result\nNUM_TDX_PRIV_KEYS: $NUM_TDX_PRIV_KEYS"
     SGX_AND_MCHECK_STATUS=$(sudo rdmsr 0xa0)
     result="$result\nSGX_AND_MCHECK_STATUS: $SGX_AND_MCHECK_STATUS (expected value: 0)"
+    PROD=$(rdmsr 0xce -f 27:27)
```

Comment:
I think this is redundant since we're already doing this by running the `check-production.sh` script [here](https://github.com/canonical/tdx/blob/558594ea444286bf198a26df1acdbe9d2d9895e2/system-report.sh#L109-L111). So either this should (a) be moved to the section below and we should not be running `check-production.sh` in that section, or (b) we should remove the dedicated section for the production checking. I'm fine either way. 

I suppose another option would be to just migrate the whole `check-production.sh` script contents into `system-check.sh` and remove `check-production.sh` entirely. Although maybe it's nice to only check for production with all the other details about the system.

Reply:
@frenchwr Ah; i did not get that we already call `check-production.sh` script when i first read the issue #192

Reply:
I asked the issue reporter for more clarification but in the meantime, i will go for option b) for now to make the script as standalone as possible

Reply:
@frenchwr I was not aware that there is a discussion here about this topic. Please see issue #192 for my comments about the `check-production.sh` script.

Reply:
@BFuhry Thanks, I was following along. Sometimes I lurk. :)